sentry-spring-boot-starter/src/main/java/io/sentry/spring/boot/jdbc/SentryDsProxyAutoConfiguration.java

```diff
@@ -0,0 +1,31 @@
+package io.sentry.spring.boot.jdbc;
+
+import com.jakewharton.nopen.annotation.Open;
+import io.sentry.IHub;
+import io.sentry.dsproxy.SentryQueryExecutionListener;
+import io.sentry.spring.boot.SentryTracingCondition;
+import javax.sql.DataSource;
+import org.jetbrains.annotations.NotNull;
+import org.springframework.boot.autoconfigure.AutoConfigureAfter;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.jdbc.DataSourceAutoConfiguration;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Conditional;
+import org.springframework.context.annotation.Configuration;
+
+/** Auto-configures datasource-proxy related beans. */
+@Open
+@Configuration
+@AutoConfigureAfter(DataSourceAutoConfiguration.class)
+@ConditionalOnClass({DataSource.class, SentryQueryExecutionListener.class})
+@Conditional(SentryTracingCondition.class)
+@ConditionalOnBean(IHub.class)
```

Comment:
does this work with GraalVM? asking because of this changes https://github.com/getsentry/sentry-java/pull/1329

Reply:
This should not affect GraalVM compatibility (although there is another issue with GraalVM https://github.com/getsentry/sentry-java/issues/1373)